### ISSUE TITLE
it would be great to have a way to set seed in projector to get same UMAP/t-SNE visualization on every run

### ISSUE THREAD

**Issue by thevasudevgupta** [tensorflow/tensorboard#6161]
Please describe the problem that you’re facing and the enhancements that
you’d like to see. Feel free to include screenshots or code samples.

Currently, tensorboard produces different embedding projections on every run. But it would be cool to pass some seed while running `tensorboard --logdir ...` to make tensorboard produce the same graphs for UMAP & t-SNE on every run.

**Comment by japie1235813** [comment]
This could be fired as FR but tbh it's unlikely to be in our priority as this is for projector plugin only and for a really specific usage. We welcome contribution, tho :) 

In the meanwhile, pardon my unfamiliarity of projector, could you elaborate more about the relations between seed and the graph? Also what do you mean by "the same graph" on every run? I image that means on dimension reduction, t-SNE and UMAP need a seed to preform the reductions and in current TB we use different seed on every run?
Also could you justify and/or provide example on how that might help and what does that mean to you? Both "the same graph" and the same seed instead of different seed for every run)

**Comment by thevasudevgupta** [comment]
> I image that means on dimension reduction, t-SNE and UMAP need a seed to preform the reductions and in current TB we use different seed on every run?

yes, and that's why I am unable to get the same projections on every run.

> Also could you justify and/or provide example on how that might help and what does that mean to you? Both "the same graph" and the same seed instead of different seed for every run)

I have been trying to find some patterns in the projections for last few days. But the main issue is that I don't find the same patterns when I run tensorboard on 2 different days. If we can fix the seed, it would help me continue my work from the state I started. Also, I would be able to use tensorboard in the live presentation because I would know graphs would be exactly same as before.

thanks!
